### PR TITLE
Tweener replacement v1

### PR DESCRIPTION
**Pull request by spotco**
not done yet

### CODE REVIEW

File: CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+end
+
+function CurveUtil:Expty(start, to, pct, dt_scale)	
+	--y = e ^ (-a * timescale)--
+	local friction = 1 - pct
+	local a = -math.log(friction)
+	return 1 - math.exp(-a * dt_scale)
+end
+
+function CurveUtil:BezierValForT(p0, p1, p2, p3, t)
```
Comment (SolarCrane) [inline CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua]: This seems like a fairly computationally expensive transformation, is this something that is used much?

File: CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+function CurveUtil:YForPointOf2PtLine(pt1, pt2, x)
+	--(y - y1)/(x - x1) = m--
+	local m = (pt1.y - pt2.y) / (pt1.x - pt2.x)
+	--y - mx = b--
+	local b = pt1.y - m * pt1.x
+	return m * x + b
+end
+
+function CurveUtil:DeltaTimeToTimescale(s_frame_delta_time)
+	return s_frame_delta_time / (1.0 / 60.0)
```
Comment (SolarCrane) [inline CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua]: I'm not sure what these conversion functions are being used for. It seems a little odd that we need to hard-code 60 here especially if we ever change the frequency that we operate at.

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua [hunk 3]
```diff
@@ -317,53 +318,86 @@ function methods:SetChannelTarget(targetChannel)
...
-	self.BackgroundTweener:CancelTween()
-	self.TextTweener:CancelTween()
-
-	self.BackgroundTweener = moduleTransparencyTweener.new()
-	self.TextTweener = moduleTransparencyTweener.new()
-
-	--// Register BackgroundTweener objects and properties
-	self.BackgroundTweener:RegisterTweenObjectProperty(self.GuiObject, "BackgroundTransparency")
-	self.BackgroundTweener:RegisterTweenObjectProperty(self.GuiObjects.TextBoxFrame, "BackgroundTransparency")
+	-- self.BackgroundTweener:CancelTween()
```
Comment (SolarCrane) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: There is a lot of commented out code in this file, if we are no longer using it we should delete this dead-code. If we ever need this code again we can use revision history.

File: CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua (hunk 2 above)
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua]: especially because we sort of already do, vive renders at 90.

Comment (spotco) [inline CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua]: @SolarCrane  actually, it's not that the "animations run at 60fps". The animation run rate is variable (determined by the master loop). The constants within your application code are set against a "60fps" animation.

File: CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua (hunk 3 above)
Comment (spotco) [inline CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua]: all the commented out stuff will be deleted. this is work in progress

File: CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua (hunk 1 above)
Comment (spotco) [inline CoreScriptsRoot/Modules/Server/ClientChat/CurveUtil.lua]: havent done all the animations yet

**Comment by hockeychaos** [comment]
do we still need this now that conor is working on his own?

**Comment by spotco** [comment]
delet this

**Pull request closed without merging**